File: Problemas/510A - Fox And Snake/510A_Fox_And_Snake.py
# 510A - Fox and Snake 
# https://codeforces.com/problemset/problem/510/A

# Input:
# 1) n = Filas
# 2) m = Columnas
n_filas, m_columnas = map(int, input().split())

# Construye la serpiente
for fila in range(1, n_filas+1):

    # Imprime la serpiente moviéndose en filas impares
    if fila % 2 == 1:
        
        # Llena la fila de #
        for columa in range(1, m_columnas+1):
            print('#', end="")

    # Imprime la serpiente moviéndose del lado derecho
    if fila % 2 == 0 and fila % 4 != 0:
         
        # Llena la fila de . excepto la última columna
        for columa in range(1, m_columnas+1):
            if columa == m_columnas:
                print('#', end="")
            else:
                print('.', end="")
    
    # Imprime la serpiente moviéndose del lado izquierdo
    if fila % 4 == 0:
         
        # Llena la fila de . excepto la primera columna
        for columa in range(1, m_columnas+1):
            if columa == 1:
                print('#', end="")
            else:
                print('.', end="")

    print('')
        

File: Problemas/510A - Fox And Snake/510A_Fox_And_Snake.cpp
// 510A - Fox and Snake 
// https://codeforces.com/problemset/problem/510/A

#include <stdio.h>
#include <stdlib.h>
#include <bits/stdc++.h>
#include <cmath>
using namespace std;

int main(){

    // Input:
    // 1) n_filas = Número de filas en el tablero
    // 2) m_columnas = Número de columnas en el tablero
    int n_filas, m_columnas;
    cin >> n_filas >> m_columnas;

    // Construye la serpiente
    for(int fila = 1; fila <= n_filas; fila++) {

        // Imprime la serpiente moviéndose en filas impares
        if(fila % 2 == 1) {

            // Llena la fila de #
            for(int columna = 1; columna <= m_columnas; columna++) {
                cout << "#";
            }
        }

        // Imprime la serpiente moviéndose del lado derecho
        if(fila % 2 == 0 && fila % 4 != 0) {

            // Llena la fila de puntos excepto en la última columna
            for(int columna = 1; columna <= m_columnas; columna++) {
                if(columna == m_columnas) {
                    cout << "#";
                } else {
                    cout << ".";
                }
            }
        }

        // Imprime la serpiente moviéndose del lado derecho
        if(fila % 4 == 0) {

            // Llena la fila de puntos excepto en la última columna
            for(int columna = 1; columna <= m_columnas; columna++) {
                if(columna == 1) {
                    cout << "#";
                } else {
                    cout << ".";
                }
            }
        }

        cout << endl;
    }

    return 0;
}
